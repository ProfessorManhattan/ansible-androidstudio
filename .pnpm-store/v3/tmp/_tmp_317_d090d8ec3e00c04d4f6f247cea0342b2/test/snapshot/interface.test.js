const parser = require("../main");

describe("interface", function () {
  it("interface name as identifier", function () {
    expect(parser.parseEval("interface A {}")).toMatchSnapshot();
  });
  it("extends", function () {
    expect(parser.parseEval("interface A extends B {}")).toMatchSnapshot();
  });
  it("multiple extends", function () {
    expect(parser.parseEval("interface A extends B, C {}")).toMatchSnapshot();
  });
});
