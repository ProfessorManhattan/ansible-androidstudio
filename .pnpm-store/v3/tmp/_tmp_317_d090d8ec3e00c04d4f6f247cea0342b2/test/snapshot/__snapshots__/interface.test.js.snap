// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`interface extends 1`] = `
Program {
  "children": Array [
    Interface {
      "attrGroups": Array [],
      "body": Array [],
      "extends": Array [
        Name {
          "kind": "name",
          "name": "B",
          "resolution": "uqn",
        },
      ],
      "kind": "interface",
      "name": Identifier {
        "kind": "identifier",
        "name": "A",
      },
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`interface interface name as identifier 1`] = `
Program {
  "children": Array [
    Interface {
      "attrGroups": Array [],
      "body": Array [],
      "extends": null,
      "kind": "interface",
      "name": Identifier {
        "kind": "identifier",
        "name": "A",
      },
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`interface multiple extends 1`] = `
Program {
  "children": Array [
    Interface {
      "attrGroups": Array [],
      "body": Array [],
      "extends": Array [
        Name {
          "kind": "name",
          "name": "B",
          "resolution": "uqn",
        },
        Name {
          "kind": "name",
          "name": "C",
          "resolution": "uqn",
        },
      ],
      "kind": "interface",
      "name": Identifier {
        "kind": "identifier",
        "name": "A",
      },
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;
